pandas/core/reshape/pivot.py

```diff
@@ -368,18 +368,34 @@ def _convert_by(by):
 @Appender(_shared_docs['pivot'], indents=1)
 def pivot(data, index=None, columns=None, values=None):
     if values is None:
-        cols = [columns] if index is None else [index, columns]
+        # Make acceptable for multiple column indexes.
+        # cols = []
+        if is_list_like(index):
+            cols = index  # cols.extend(index)
+        elif index is not None:
+            cols = [index]
+        else:
+            cols = []
+        cols.append(columns)
+
         append = index is None
         indexed = data.set_index(cols, append=append)
+
     else:
         if index is None:
-            index = data.index
+            index = MultiIndex.from_arrays([data.index, data[columns]])
+        elif is_list_like(index):
```

Comment:
I think it would help readability if the conditions here were refactored to be in the same sequence as the conditions in the branch above, as IIUC we essentially evaluate the same conditions in both branches